connectors/utils.py

```diff
@@ -300,3 +300,31 @@ async def _wait_for_room(self, item):
     async def put(self, item):
         item_size = await self._wait_for_room(item)
         return await super().put((item_size, item))
+
+
+class ConcurrentRunner:
+    def __init__(self, max_concurrency=5, results_cb=None):
+        self.max_concurrency = 5
+        self.tasks = []
+        self.results_cb = results_cb
+
+    def __len__(self):
+        return len(self.tasks)
+
+    def _cb(self, fut):
+        self.tasks.remove(fut)
+        if self.results_cb is not None:
+            if fut.exception():
+                raise fut.exception()
+            self.results_cb(fut.result())
+
+    async def put(self, coro):
+        while len(self.tasks) >= self.max_concurrency:
+            await asyncio.sleep(0)
```

Comment:
Curious about that part - is it gonna destroy CPU this way?

Reply:
Great question!

It will return immediately to the event loop, and the assumption is that since the other coroutines are busy pushing data to Elastic or retrieving some from the backend, the CPU should be fine. But in theory you are absolutely right, if we keep on coming back here it'll destroy the CPU.

A much better pattern would be to unblock it through `asyncio.Event`. e.g. another task was finished. I will do this :)